openllm-core/src/openllm_core/prompt/prompt_template.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+import typing as t
+
+import attr
+
+from .utils import default_formatter
+
+@attr.define(slots=True)
+class PromptTemplate:
+  prompt_template: str
+  input_variables: t.Mapping[str, str]
+  prepared_template: str = attr.field(init=False)
+
+  def __attrs_post_init__(self) -> None:
+    template_variables = default_formatter.extract_template_variables(self.prompt_template)
+    prompt_variables = {key: '{' + key + '}' if key not in self.input_variables else self.input_variables[key] for key in template_variables}
+    self.prepared_template = self.prompt_template.format(**prompt_variables)
+
+  def with_input_variables(self, input_variables: t.Mapping[str, str]) -> str:
+    template_variables = default_formatter.extract_template_variables(self.prompt_template)
+    merged_inputs = {**self.input_variables, **input_variables}
+    prompt_variables = {key: '{' + key + '}' if key not in merged_inputs else merged_inputs[key] for key in template_variables}
+    return self.prompt_template.format(**prompt_variables)
```

Comment:
refactor process_prompt to PromptTemplate.format

Reply:
process_template still lives on for now until prompt template is integrated to the rest of the models